src/govuk/components/character-count/character-count.js

```diff
@@ -50,6 +50,20 @@ CharacterCount.prototype.init = function () {
   // Remove hard limit if set
   $module.removeAttribute('maxlength')
 
+  // When the page is restored after navigating 'back' in some browsers the
+  // state of the character count is not restored until *after* the DOMContentLoaded
+  // event is fired, so we need to sync after the pageshow event in browsers
+  // that support it.
+  if ('onpageshow' in window) {
+    window.addEventListener('pageshow', this.syncAll.bind(this))
+  } else {
+    window.addEventListener('DOMContentLoaded', this.syncAll.bind(this))
+  }
+
+  this.syncAll()
+}
+
+CharacterCount.prototype.syncAll = function () {
```

Comment:
Wondering if `syncAll` is the right name for this function, as unlike the radios and checkboxes there is only one character count per 'module', so there's only really one thing to 'sync'… thoughts?